stdlib/src/array.rs

```diff
@@ -215,8 +215,10 @@ mod array {
                             // support are valid
                             let ptr = b.as_ptr() as *const $t;
                             let ptr_len = b.len() / std::mem::size_of::<$t>();
-                            let slice = unsafe { std::slice::from_raw_parts(ptr, ptr_len) };
-                            v.extend_from_slice(slice);
+                            if ptr_len > 0 {
```

Comment:
```suggestion
                                if b.len() > 0 {
                                    let ptr = b.as_ptr() as *const $t;
                                    let ptr_len = b.len() / std::mem::size_of::<$t>();
                                    let slice = unsafe { std::slice::from_raw_parts(ptr, ptr_len) };
```

This is a little bit weird though. Because extend_from_slice have not to copy anything from the slice if its length is 0.